src/utils/private.ts

```diff
@@ -92,7 +92,7 @@ export const getPrivateKey = async (cipherText: string): Promise<string | undefi
     const binCipher = sodium.from_base64(cipherText, sodium.base64_variants.URLSAFE_NO_PADDING);
 
     let walletPrivateKey: string | undefined = sodium.crypto_box_seal_open(binCipher, binPub, binPriv, "text");
-    walletPrivateKey = walletPrivateKey.startsWith(KEY_PREFIX) ? walletPrivateKey.replace(KEY_PREFIX, "") : undefined;
```

Comment:
Check an example from a partner's view:
1. Partner encrypts **manually** his private key without the `HSK` prefix
2. On the bot's side on permit generation `walletPrivateKey` is `undefined` because there is no `HSK` prefix. 

I don't get what is `the rest of the logic`. Then this PK is used here simply reading from env https://github.com/ubiquity/ubiquibot/blob/development/src/utils/private.ts#L129

So I'm trying to cover a case when there is no `HSK` prefix.

Reply:
I see, although partners shouldn't use raw libsodium to encrypt the keys (they should use the frontend)